src/App.js

```diff
@@ -0,0 +1,48 @@
+import { Route, Routes } from "react-router-dom";
+import Forget from "./pages/Forget/index.jsx";
+import Home from "./pages/Home";
+import LoginPage from "./pages/LoginPage";
+import ProductItems from "./pages/ProductItems/index";
+import Register from "./pages/Register/index";
+import SingleProduct from "./pages/SingleProduct/index.jsx";
+import Cart from "./pages/Cart/index";
+import AllProducts from "./components/AllProducts/";
+import { MainContext } from "./Context";
+import { useContext, useEffect } from "react";
+import { ToastContainer } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import Wishlist from "./pages/Wishlist";
+
+function App() {
+  const { state, dispatch } = useContext(MainContext);
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const res = await fetch("/api/products", { method: "GET" });
+        const data = await res.json();
+        dispatch({ type: "SET_PRODUCTS", payload: data.products });
+      } catch (e) {
+        console.log(state);
+      }
+    }
+    fetchData();
+  }, []);
```

Comment:
dispatch must be the dependency in useEffect

Reply:
added this change